gen/KubernetesWatchGenerator/Program.cs

```diff
@@ -0,0 +1,250 @@
+using Newtonsoft.Json;
+using NJsonSchema;
+using NSwag;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace KubernetesWatchGenerator
+{
+    class Program
+    {
+        static async Task Main(string[] args)
+        {
+            // Download the latest spec
+            const string kubernetesBranch = "v1.10.0";
+            const string outputDirectory = "../../../../../src/KubernetesClient/generated/";
+
+            // We skip operations where the name of the class in the C# client could not be determined correctly.
+            // That's usually because there are different version of the same object (e.g. for deployments).
+            Collection<string> blacklistedOperations = new Collection<string>()
+            {
+                "watchAppsV1beta1NamespacedDeployment",
+                "watchAppsV1beta2NamespacedDeployment",
+                "watchExtensionsV1beta1NamespacedDeployment",
+                "watchExtensionsV1beta1NamespacedNetworkPolicy",
+                "watchPolicyV1beta1PodSecurityPolicy",
+                "watchExtensionsV1beta1PodSecurityPolicy"
+            };
+
+            var specUrl = $"https://raw.githubusercontent.com/kubernetes/kubernetes/{kubernetesBranch}/api/openapi-spec/swagger.json";
+            var specPath = $"{kubernetesBranch}-swagger.json";
+
+            if (!File.Exists(specPath))
+            {
+                HttpClient client = new HttpClient();
+                using (var response = await client.GetAsync(specUrl, HttpCompletionOption.ResponseHeadersRead))
+                using (var stream = await response.Content.ReadAsStreamAsync())
+                using (var output = File.Open(specPath, FileMode.Create, FileAccess.ReadWrite))
+                {
+                    await stream.CopyToAsync(output);
+                }
+            }
+
+            var swagger = await SwaggerDocument.FromFileAsync(specPath);
+
+            var watchOperations = swagger.Operations.Where(
+                o => o.Path.Contains("/watch/")
+                && o.Operation.ActualParameters.Any(p => p.Name == "name")
+                && !blacklistedOperations.Contains(o.Operation.OperationId)).ToArray();
+
+            using (Stream interfaceFile = File.Open($"{outputDirectory}IKubernetes.Watch.cs", FileMode.Create, FileAccess.ReadWrite))
+            using (StreamWriter interfaceWriter = new StreamWriter(interfaceFile, Encoding.UTF8))
+            using (Stream classFile = File.Open($"{outputDirectory}Kubernetes.Watch.cs", FileMode.Create, FileAccess.ReadWrite))
+            using (StreamWriter classWriter = new StreamWriter(classFile, Encoding.UTF8))
+            {
+                interfaceWriter.WriteLine("using k8s.Models;");
```

Comment:
Rather than this can we use a templating library, for example this one:

https://github.com/wildbit/mustachio